proxysql_galera_checker

```diff
@@ -1,63 +1,162 @@
+  go_out="$(getopt --options=w:r:c:l:n:vh --longoptions=write-hg:,read-hg:,config-file:,log:,writer-count:,version,help \
+  --name="$(basename "$0")" -- "$@")"
+  test $? -eq 0 || exit 1
+  eval set -- "$go_out"
+fi
 
+if [[ $go_out == " --" ]];then
+  usage
+  exit 1
+fi
 
-Notes about the mysql_servers in ProxySQL:
+for arg
+do
+  case "$arg" in
+    -- ) shift; break;;
+    -w | --write-hg )
+      export WRITE_HG="$2"
+      shift 2
+    ;;
+    -r | --read-hg )
+      export READ_HG="$2"
+      shift 2
+    ;;
+    --config-file )
+      CONFIG_FILE="$2"
+      shift 2
+      if [ -z "${CONFIG_FILE}" ]; then
+        echo "ERROR: The configuration file location (--config-file) was not provided. Terminating."
+        exit 1
+      fi
+      if [ -e "${CONFIG_FILE}" ]; then
+        # Loading configuration from ${CONFIG_FILE}
+        check_permission "${CONFIG_FILE}"
+        source "${CONFIG_FILE}"
+      else
+        echo "ERROR: The configuration file ${CONFIG_FILE} specified by --config-file does not exist. Terminating."
+        exit 1
+      fi
+    ;;
+    -l | --log )
+      export ERR_FILE="$2"
+      shift 2
+    ;;
+    -n | --writer-count )
+      export NUMBER_WRITERS="$2"
+      shift 2
+    ;;
+    -v | --version )
+      echo "proxysql_galera_checker version 1.4.9"
+      exit 0
+    ;;
+    -h | --help )
+      usage
+      exit 0
+    ;;
+  esac
+done
 
-- WEIGHT           Hosts with a higher weight will be prefered to be put ONLINE
-- NODE STATUS      * Nodes that are in status OFFLINE_HARD will not be checked nor will their status be changed
-                   * SHUNNED nodes are not to be used with Galera based systems, they will be checked and status
-                     will be changed to either ONLINE or OFFLINE_SOFT.
+if [[ -z $CONFIG_FILE ]]; then
+  # Reading default variables from default configuration file location.
+  if [ -e "/etc/proxysql-admin.cnf" ]; then
+    # Loading default configuration from /etc/proxysql-admin.cnf
+    check_permission  "/etc/proxysql-admin.cnf"
+    source /etc/proxysql-admin.cnf
+    CONFIG_FILE="/etc/proxysql-admin.cnf"
+  else
+    echo "ERROR! Default configuration file (/etc/proxysql-admin.cnf) does not exist. Terminating"
+	exit 1
+  fi
+fi
 
+if [[ -z $ERR_FILE ]]; then
+  ERR_FILE="/dev/null"
+fi
+echoit(){
+  echo "[$(date +%Y-%m-%d\ %H:%M:%S)] $1" >> $ERR_FILE
+}
 
-When no nodes were found to be in wsrep_local_state=4 (SYNCED) for either
-read or write nodes, then the script will try 5 times for each node to try
-to find nodes wsrep_local_state=4 (SYNCED) or wsrep_local_state=2 (DONOR/DESYNC)
+if [[ -z "$PROXYSQL_DATADIR" ]]; then
+  PROXYSQL_DATADIR='/var/lib/proxysql'
+fi
 
-This is to avoid $0 to mark all nodes as OFFLINE_SOFT
+#Timeout exists for instances where mysqld/proxysql may be hung
+TIMEOUT=5
+# DEFAULTS
+if [[ -z $HOSTGROUP_WRITER_ID ]]; then
+  HOSTGROUP_WRITER_ID="1"
+else
+  if [[ ! -z $WRITE_HG ]]; then
+    HOSTGROUP_WRITER_ID=$WRITE_HG
+  fi
+fi
```

Comment:
@rameshvs02 if the sourced file has no `HOSTGROUP_WRITER_ID` but we passed `--write-hg` as part of the scheduler call, it will assume `HOSTGROUP_WRITER_ID=1` instead of whatever we passed on `--write-hg`